### PR TITLE
Moon Outpost 19 mediumcore rework:

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Moon Outpost 19, the middle children of the away mission maps. It's not really all that spectacular, but at the same time, the map file has a colossal amount of empty, blank space. As opposed to just removing it, I did some sprites, tweaks, and mapping changes to get a good old fashion update to a less beloved map to give it just a little bit of soul.

New things:
**Lunar Sand!** 
Unlike the old yellow colored ocean sand that we had on the outpost, the base now has lunar-accurate solar-washed gray sand. It behaves in much the same way as regular basalt or yellow sand, but with more of a rocky texture and darker hanging shadows over the rocky surface.

**Alien Plants and Weeds!** 
Uniquely, the caves and xenoflora of the moon outpost have grown over the xenomorph weeds, giving a tiny, unique bit of diversity to the usual xenoflora of weeds, nests, and nodes. 

**Misc Objects and Sprites!** 
Fluff meteorites have been added as static objects, not spinning and hurdling at you for a coup de gras, but instead as a fluff object that has landed and being studied by the moon outpost 19 staff. Additionally, some new rock sprites were also added to the random rock pool to help diversify the map pool since the rocky outcrop of the moon features a lot of... rocks.

**Soul Pass and Var Edits!** 
Some meticulous detail was used in overseeing the map and the objects contained therein. This includes some custom names, descriptions, lots of unique placement, and better atmospheric storytelling of the tragedy that befell this research outpost before the xenomorph attack was underway. This is primarily an expansion of the existing rooms and nooks and crannies of the old map as well as some of the new rooms added.

**Oh yeah new rooms and areas were added!** 
This map was already a 255x255 tile space, but very, VERY little of the map was used before it was originally implemented. Now, we've added a few new rooms and challenges to the map to help in both the flavor of the map and giving players more to do on the map itself. The main 3 areas added are:
 * The ATV Shed, used for extended outdoor missions on the moon outpost away from the research station. Seems like the shed was locked up tight after the last office party however, and the late RD was the last one to have the key...
 * Ground Zero, the original excavation site, where basalt and rocks from Lavaland broke off and landed on the moon, resulting in an investigation as these rocks were thousands of years older than modern deposits from Lavaland. However, it seems that something found it's way back as well.
 * The Lunar Caves, connecting the northern and southern sites together. The monsters appear to have set up a forward base here, catching the remaining stragglers off guard. Will you be the next?

****

## Why It's Good For The Game

This away mission wasn't in a very good place beforehand. I started working on this a few months ago, and planned a much more wide-spread and elaborate away mission rework. However, due to A: the away mission bounty not covering existing maps and B: Me having to dedicate almost all my time to academics a few months ago, I really lost all my steam to change this map up. That said, I remembered that I had all this work done already, and after fixing some minor issues wanted to make sure that what I had was pushed to the main game. Obviously, there's a great deal more work that could be done on this map and I'm open to suggestions as well as collaboration, but for the moment this really adds a great deal of flavor to an otherwise rather unremarkable away mission in rotation.

I'd say it's a mediumcore rework because, sadly, not a lot of the core gameplay has changed, but the aesthetics of the map should have seen an impressive improvement.

## Changelog

:cl:
add: Moon Outpost 19's away mission has been expanded upon! Investigate the ruins of an old Nanotrasen research station.
imageadd: Lunar rock, and xenofauna sprites have been added.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -818,6 +818,26 @@ GLOBAL_LIST_EMPTY(flora_uprooting_tools_typepaths)
...
+/obj/structure/flora/lunar_plant/Initialize(mapload)
+	. = ..()
+	icon_state = "cave_plant[rand(1,3)]"
+
+/obj/structure/flora/lunar_plant/style_1
+	icon_state = "lunar_plant1"
+/obj/structure/flora/lunar_plant/style_2
+	icon_state = "lunar_plant2"
+/obj/structure/flora/lunar_plant/style_3
+	icon_state = "lunar_plant3"
```
Comment (san7890) [inline code/game/objects/structures/flora.dm:839]:
```suggestion
/obj/structure/flora/lunar_plant/style_1
	icon_state = "lunar_plant1"

/obj/structure/flora/lunar_plant/style_2
	icon_state = "lunar_plant2"

/obj/structure/flora/lunar_plant/style_3
	icon_state = "lunar_plant3"
```

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -838,14 +858,35 @@ GLOBAL_LIST_EMPTY(flora_uprooting_tools_typepaths)
...
+	SSblackbox.record_feedback("tally", "pick_used_mining", 1, attacking_item.type)
+	qdel(src)
 
 /obj/structure/flora/rock/style_2
 	icon_state = "basalt2"
 /obj/structure/flora/rock/style_3
 	icon_state = "basalt3"
+/obj/structure/flora/rock/style_4
+	icon_state = "basalt4"
 /obj/structure/flora/rock/style_random/Initialize(mapload)
```
Comment (san7890) [inline code/game/objects/structures/flora.dm:887]:
```suggestion
/obj/structure/flora/rock/style_2
	icon_state = "basalt2"

/obj/structure/flora/rock/style_3
	icon_state = "basalt3"

/obj/structure/flora/rock/style_4
	icon_state = "basalt4"

/obj/structure/flora/rock/style_random/Initialize(mapload)
```

here too would be nice

File: code/game/objects/structures/flora.dm [hunk 3]
```diff
@@ -838,14 +858,35 @@ GLOBAL_LIST_EMPTY(flora_uprooting_tools_typepaths)
 	flora_flags = FLORA_STONE
 	can_uproot = FALSE
 	delete_on_harvest = TRUE
+	rock = TRUE
+	/// Itemstack that is dropped when a rock is mined with a pickaxe
+	var/obj/item/stack/mineResult = /obj/item/stack/ore/glass/basalt
+	/// Amount of the itemstack to drop
+	var/mineAmount = 20
+
+/obj/structure/flora/rock/attackby(obj/item/attacking_item, mob/user, params)
```
[the remark this answers is not in the document]
Comment (ArcaneMusic) [reply code/game/objects/structures/flora.dm:869]: Oh hold on, no, it's worse, someone refactored flora a month ago and I wasn't aware of that. Time to axe a good part of this.

File: code/game/objects/structures/flora.dm (hunk 1 above)
Comment (ArcaneMusic) [reply code/game/objects/structures/flora.dm]: Everything in this file lacks newlines it would seem. I'll take this as reason to fix that.

**Comment by Edelor68** [comment]
Pictures?

**Comment by Paxilmaniac** [comment]
image
It looks like this has a stray white pixel

**Comment by ArcaneMusic** [comment]
> Pictures?

Pictures are going to be rather difficult to snag due to how large the map is, but an example of the initial base edits is here:
image

For a more complete view, map diff bot is your friend.

**Review by san7890 — CHANGES_REQUESTED** [review]
According to the path forward on Gateways Design Doc: https://hackmd.io/U-QeMp5qT_yyw1Cx2Q6U3Q?view

image

This seems to address those two points. Great! Let's go through some things. I unfortunately do not have much time tonight (was expecting to have more), but let's get a ball rolling. The time constraint is that I don't have much time to go through and run a debug server on this away mission, great apologies. We'll do a second review later on (hopefully not too far in the future).

<details>
<summary>Expand Me Here</summary>

image

Is this power line meant to be disconnected? It looks like it should be connected, I think you can crowbar depowered shutters? I could be wrong about that, but it's kinda hard to charge the APC if you can't... get under the shutters to connect the cable. I'd much prefer if it ran under the keycard door anyways.

image
Love this usage of notice boards by the way.

image
This little hut shares the same area as the rest of the station, but they both have unique APCs. I would give the hut it's own area.

image
Feel free to have some fun with the lore, name the display case/trophy something nice (if you want).

image
This name is probably good for like... something, but I think the parenthesis really take you out of it all. I know it's how it is in the code, but it would be nice if you could rename it such that it's not just obscenely OOC.

image

The mineral is already indestructible, why make it five tiles thick? This is definitely a holdover from it's initial iterations, one that I don't really think should be needed (especially since we should have baseturf helpers here). Actually, speaking of base turf helpers, I didn't see one for some of the areas, could you please do another pass on that for me? You might need to make a new one for your cool new moon sprites.

image

If you want brownie points, it would be very cool and nice if you could "flatten" the sprites using stuff like anticorner/half(contrasted). Not required, but would be very nice.

image

Should this room remain reinforced?

image

This looks _really_ bad. Dark four corners were bad, having two over a white floor looks worse. It's just really bizarre and I think it might be nicer to use normal dark flooring here isntead?

image
This catwalk is missing the dust the others have, intentional? I _personally_ think it should all be consistent to give it the dilapidated feel, but it's up to you.

image
This whole section has a really bizarre mix of indestructible/normal rock. Like... a thin line of indestructible stuff? It's hard to tell because I keep losing the line. I'd rather have the edges be indestructible if that's what you're going for, such that the player instantly knows that it's not normal. Maybe one normal then a wall of indestructible? It's hard for me to tell at a glance and I just think the less obtusity, the better.

image

What're the use of these cables?
</details>

That's all I really have for you at this time, just getting the ball rolling. I'll be able to provide much more feedback when I have enough time to compile it (versus staring at it in StrongDMM). Thank you for your work thus far.

**Comment by ArcaneMusic** [comment]
Closing until I get a bit more free time, if anyone gets the urge to finish this while I'm still away they have my permission

**Pull request closed without merging**